apps/vscode-extension/src/commands/extract/styling/commons/core/vscode-env.ts

```diff
@@ -0,0 +1,191 @@
+  public get filepath(): string { return assertValue(this._filepath, 'Filepath not defined, please check also TextDocument'); }
+
+  public get forbiddenWords(): string[] {
+    if (this._overridenConfig && this._overridenConfig.forbiddenWords) {
+      return this._overridenConfig.forbiddenWords;
+    }
+    return workspace.getConfiguration('otter.extract.styling').get<string[]>('forbiddenWords') ?? [];
+  }
+
+  public get prefix(): string {
+    if (this._overridenConfig && this._overridenConfig.prefix) {
+      return this._overridenConfig.prefix;
+    }
+    return workspace.getConfiguration('otter.extract.styling').get<string>('prefix') ?? '';
+  }
+
+  public get functionImportPath(): string {
+    if (this._overridenConfig && this._overridenConfig.functionImportPath) {
+      return this._overridenConfig.functionImportPath;
+    }
+    return workspace.getConfiguration('otter.extract.styling').get<string>('functionImportPath') ?? '';
+  }
+
+  public setCursor(line: number, char: number): void {
+    this._position = new Position(line, char);
+  }
+
+  /**
+   * Lock the refresh of the VscodeEnv instance.
+   * Usefull for test purpose if a specific configuration needs to be done
+   * And to avoid the refresh during the execution of the command.
+   */
+  public lockRefresh(): void {
+    this._lockRefresh = true;
+  }
+
+  /**
+   * Unlock the refreh as explained with **lockRefresh()**.
+   */
+  public unlockRefresh(): void {
+    this._lockRefresh = false;
+  }
+}
+
+export const vscodeEnv = new VscodeEnv();
+
+/**
+ * Override the default environnement (usefull for testing purpose)
+ * Doing a factory call will lock any other refresh in the application
+ * You can unlock it calling the `vscodeEnv.unlockRefresh()` method
+ *
+ * @param opts text, activeLine, activeChar, dryRun, overrideConfig
+ * @param opts.text text to write in the active editor
+ * @param opts.activeLine active line of the cursor
+ * @param opts.activeChar active char of the cursor
+ * @param opts.dryRun block some logics (like renaming command or file extension verification)
+ * @param opts.overrideConfig configuration to be used
+ * @returns the updated instance of VscodeEnv
+ */
+export async function createFactoryVscodeEnv(
+  { text, activeLine, activeChar, dryRun, overrideConfig }: {
+    text: string;
+    activeLine?: number;
+    activeChar?: number;
+    dryRun?: boolean;
+    overrideConfig?: ExtExtractStylingConfig;
+  }
+): Promise<VscodeEnv> {
+  const document = await workspace.openTextDocument({ content: text });
+  const editor = await window.showTextDocument(document);
+  const position = activeLine && activeChar ? new Position(activeLine, activeChar) : undefined;
+  vscodeEnv.unlockRefresh();
+  const vscodeInstance = vscodeEnv.refresh({ position, document, editor, dryRun, overrideConfig });
+  vscodeEnv.lockRefresh();
+  return vscodeInstance;
+}
+
+export default vscodeEnv;
```

Comment:
Why export default ?

Reply:
Maybe the answer should comes from @JasonHaenlinAmadeus but as I do not have good justification I turn it to `export class` instead

Reply:
it was mainly because of my habit because it's a single class in his own file and not multiple functions but it can be omitted depending on your practices.